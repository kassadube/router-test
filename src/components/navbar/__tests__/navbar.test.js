/* eslint-disable react/react-in-jsx-scope */

import Navbar from '../index';

test('# Navbar at home', () => {
  const wrapperRoot = shallow(<Navbar current="/" />);
  expect(wrapperRoot).toMatchSnapshot();
});

test('# Navbar at create', () => {
  const wrapperCreate = shallow(<Navbar current="/create" />);
  expect(wrapperCreate).toMatchSnapshot();
});

test('# Navbar at with user loged in', () => {
  const wrapperUser = shallow(<Navbar user={{login: 'test'}} current="/profile/me" />);
  expect(wrapperUser).toMatchSnapshot();
});

describe('mount testing', () => {
  it('#navbar loged in user', () => {
    const component = (<Navbar user={{login: 'test'}} current="/profile/me" />);
    const wrapper = mount(component);
    expect(wrapper).toMatchSnapshot();
  });
  it('#navbar not loged in user', () => {
    const component = (<Navbar current="/profile/me" />);
    const wrapper = mount(component);
    expect(wrapper).toMatchSnapshot();
  });
});
describe('rendering testing', () => {
  it('#navbar loged in user RENDER', () => {
    const component = (<Navbar user={{login: 'test'}} current="/profile/me" />);
    const wrapper = render(component);
    expect(wrapper).toMatchSnapshot();
  });
  it('#navbar not loged in user RENDER', () => {
    const component = (<Navbar current="/profile/me" />);
    const wrapper = render(component);
    expect(wrapper).toMatchSnapshot();
  });
});
