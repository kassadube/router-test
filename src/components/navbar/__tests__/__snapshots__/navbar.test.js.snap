exports[`mount testing #navbar loged in user 1`] = `
<navbar
  current="/profile/me"
  user={
    Object {
      "login": "test",
    }
  }>
  <nav
    className="navbar navbar-default">
    <div
      className="container-fluid">
      <div
        className="navbar-header">
        <Link
          className="navbar-brand"
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/">
          <a
            className="navbar-brand"
            onClick={[Function]}
            style={Object {}}>
            Brand
          </a>
        </Link>
      </div>
      <ul
        className="nav navbar-nav">
        <li>
          <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/other">
            <a
              onClick={[Function]}
              style={Object {}}>
              page Not Found
            </a>
          </Link>
        </li>
        <li
          className="active">
          <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/">
            <a
              onClick={[Function]}
              style={Object {}}>
              Browse questions
            </a>
          </Link>
        </li>
        <li>
          <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/create">
            <a
              onClick={[Function]}
              style={Object {}}>
              Create new questions
            </a>
          </Link>
        </li>
      </ul>
      <ul
        className="nav navbar-nav navbar-right">
        <li>
          <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/profile/me">
            <a
              onClick={[Function]}
              style={Object {}}>
              test
            </a>
          </Link>
        </li>
        <li>
          <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/logout">
            <a
              onClick={[Function]}
              style={Object {}}>
              logout
            </a>
          </Link>
        </li>
      </ul>
    </div>
  </nav>
</navbar>
`;

exports[`mount testing #navbar not loged in user 1`] = `
<navbar
  current="/profile/me"
  user={Object {}}>
  <nav
    className="navbar navbar-default">
    <div
      className="container-fluid">
      <div
        className="navbar-header">
        <Link
          className="navbar-brand"
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/">
          <a
            className="navbar-brand"
            onClick={[Function]}
            style={Object {}}>
            Brand
          </a>
        </Link>
      </div>
      <ul
        className="nav navbar-nav">
        <li>
          <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/other">
            <a
              onClick={[Function]}
              style={Object {}}>
              page Not Found
            </a>
          </Link>
        </li>
        <li
          className="active">
          <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/">
            <a
              onClick={[Function]}
              style={Object {}}>
              Browse questions
            </a>
          </Link>
        </li>
        <li>
          <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/create">
            <a
              onClick={[Function]}
              style={Object {}}>
              Create new questions
            </a>
          </Link>
        </li>
      </ul>
      <ul
        className="nav navbar-nav navbar-right">
        <li>
          <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/profile/me">
            <a
              onClick={[Function]}
              style={Object {}} />
          </Link>
        </li>
        <li>
          <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/logout">
            <a
              onClick={[Function]}
              style={Object {}}>
              logout
            </a>
          </Link>
        </li>
      </ul>
    </div>
  </nav>
</navbar>
`;

exports[`rendering testing #navbar loged in user RENDER 1`] = `
<nav
  class="navbar navbar-default">
  <div
    class="container-fluid">
    <div
      class="navbar-header">
      <a
        class="navbar-brand">
        Brand
      </a>
    </div>
    <ul
      class="nav navbar-nav">
      <li>
        <a>
          page Not Found
        </a>
      </li>
      <li
        class="active">
        <a>
          Browse questions
        </a>
      </li>
      <li>
        <a>
          Create new questions
        </a>
      </li>
    </ul>
    <ul
      class="nav navbar-nav navbar-right">
      <li>
        <a>
          test
        </a>
      </li>
      <li>
        <a>
          logout
        </a>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`rendering testing #navbar not loged in user RENDER 1`] = `
<nav
  class="navbar navbar-default">
  <div
    class="container-fluid">
    <div
      class="navbar-header">
      <a
        class="navbar-brand">
        Brand
      </a>
    </div>
    <ul
      class="nav navbar-nav">
      <li>
        <a>
          page Not Found
        </a>
      </li>
      <li
        class="active">
        <a>
          Browse questions
        </a>
      </li>
      <li>
        <a>
          Create new questions
        </a>
      </li>
    </ul>
    <ul
      class="nav navbar-nav navbar-right">
      <li>
        <a />
      </li>
      <li>
        <a>
          logout
        </a>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`test # Navbar at create 1`] = `
<nav
  className="navbar navbar-default">
  <div
    className="container-fluid">
    <div
      className="navbar-header">
      <Link
        className="navbar-brand"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/">
        Brand
      </Link>
    </div>
    <ul
      className="nav navbar-nav">
      <li>
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/other">
          page Not Found
        </Link>
      </li>
      <li
        className="active">
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/">
          Browse questions
        </Link>
      </li>
      <li>
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/create">
          Create new questions
        </Link>
      </li>
    </ul>
    <ul
      className="nav navbar-nav navbar-right">
      <li>
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/profile/me" />
      </li>
      <li>
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/logout">
          logout
        </Link>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`test # Navbar at home 1`] = `
<nav
  className="navbar navbar-default">
  <div
    className="container-fluid">
    <div
      className="navbar-header">
      <Link
        className="navbar-brand"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/">
        Brand
      </Link>
    </div>
    <ul
      className="nav navbar-nav">
      <li>
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/other">
          page Not Found
        </Link>
      </li>
      <li
        className="active">
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/">
          Browse questions
        </Link>
      </li>
      <li>
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/create">
          Create new questions
        </Link>
      </li>
    </ul>
    <ul
      className="nav navbar-nav navbar-right">
      <li>
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/profile/me" />
      </li>
      <li>
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/logout">
          logout
        </Link>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`test # Navbar at with user loged in 1`] = `
<nav
  className="navbar navbar-default">
  <div
    className="container-fluid">
    <div
      className="navbar-header">
      <Link
        className="navbar-brand"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/">
        Brand
      </Link>
    </div>
    <ul
      className="nav navbar-nav">
      <li>
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/other">
          page Not Found
        </Link>
      </li>
      <li
        className="active">
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/">
          Browse questions
        </Link>
      </li>
      <li>
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/create">
          Create new questions
        </Link>
      </li>
    </ul>
    <ul
      className="nav navbar-nav navbar-right">
      <li>
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/profile/me">
          test
        </Link>
      </li>
      <li>
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/logout">
          logout
        </Link>
      </li>
    </ul>
  </div>
</nav>
`;
